tests/integration/test_integration_all_rust.py

```diff
@@ -590,983 +583,1003 @@ def teardown_module():
     kill_process(EP_SERVER)
 
 
-class TestRustWebPush:
-    """Test class for Rust Web Push."""
+@pytest.fixture(name="ws_url", scope="session")
+def fixture_ws_url() -> str:
+    """Return defined url for websocket connection."""
+    return f"ws://localhost:{CONNECTION_PORT}/"
+
+
+@pytest.fixture(name="broadcast_ws_url", scope="session")
+def fixture_broadcast_ws_url() -> str:
+    """Return websocket url for megaphone broadcast testing."""
+    return f"ws://localhost:{MP_CONNECTION_PORT}/"
+
 
-    # Max log lines allowed to be emitted by each node type
-    max_endpoint_logs = 8
-    max_conn_logs = 3
-    vapid_payload = {
+@pytest.fixture(name="vapid_payload", scope="session")
+def fixture_vapid_payload() -> dict[str, int | str]:
+    """Return a test fixture of a vapid payload."""
+    return {
         "exp": int(time.time()) + 86400,
         "sub": "mailto:admin@example.com",
     }
 
-    @staticmethod
-    def clear_sentry_queue() -> None:
-        """Clear any values present in Sentry queue."""
-        while not MOCK_SENTRY_QUEUE.empty():
-            MOCK_SENTRY_QUEUE.get_nowait()
-
-    def tearDown(self):
-        """Tear down and log processing."""
-        process_logs(self)
-        TestRustWebPush.clear_sentry_queue()
-
-    def host_endpoint(self, client):
-        """Return host endpoint."""
-        parsed = urlparse(list(client.channels.values())[0])
-        return f"{parsed.scheme}://{parsed.netloc}"
-
-    async def quick_register(self):
-        """Perform a connection initialization, which includes a new connection,
-        `hello`, and channel registration.
-        """
-        log.debug(f"🐍#### Connecting to ws://127.0.0.1:{CONNECTION_PORT}/")
-        client = AsyncPushTestClient(f"ws://127.0.0.1:{CONNECTION_PORT}/")
-        await client.connect()
-        await client.hello()
-        await client.register()
-        log.debug("🐍 Connected")
-        return client
-
-    async def shut_down(self, client=None):
-        """Shut down client."""
-        if client:
-            await client.disconnect()
-
-    @property
-    def _ws_url(self):
-        return f"ws://127.0.0.1:{CONNECTION_PORT}/"
-
-    @pytest.mark.asyncio
-    @max_logs(conn=4)
-    async def test_sentry_output_autoconnect(self):
-        """Test sentry output for autoconnect."""
-        if os.getenv("SKIP_SENTRY"):
-            SkipTest("Skipping sentry test")
-            return
-        # Ensure bad data doesn't throw errors
-        client = AsyncPushTestClient(self._ws_url)
-        await client.connect()
-        await client.hello()
-        await client.send_bad_data()
+
+@pytest.fixture(name="clear_sentry_queue", autouse=True, scope="function")
+def fixture_clear_sentry_queue():
```

Comment:
nit
```suggestion
def fixture_clear_sentry_queue() -> None:
```

Reply:
Due to use of `yield`, this is actually a `Generator`, which is fine to know but not super helpful... thanks mypy haha